Add Vertex Input Data Processing chapter (#62)
KhronosGroup/Vulkan-Guide: chapters/vertex_input_data_processing.md

# Vertex Input Data Processing

This chapter is an overview of the [Fixed-Function Vertex Processing chapter in the spec](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#fxvertex) to help give a high level understanding of how an application can map data to the vertex shader when using a graphics pipeline.

> It is also important to remember that Vulkan is a tool with many possible ways to use. The following are examples for educational purposes of how vertex data **can** be laid out.

Reviewer: ```suggestion
It is also important to remember that Vulkan is a tool that can be used in different ways. The following are examples for educational purposes of how vertex data **can** be laid out.
```

```patch
        1,                          // location
        binding.binding,            // binding
        VK_FORMAT_R8G8_UNORM,       // format
-        3 * sizeof(float)           // offset
+        4 * sizeof(float)           // offset
```

![vertex_input_data_processing_example_b_offset](../images/vertex_input_data_processing_example_b_offset.png)

#### Adjusting format

For the second case changing the format, you keep the `3 * sizeof(float)` offset and adjust the format to change the size of each location element

```patch
        0,                          // location
        binding.binding,            // binding
-        VK_FORMAT_R32G32B32_SFLOAT, // format
+        VK_FORMAT_R32G32B32A32_SFLOAT, // format
        0                           // offset
```

Note, this does **not** require an update to the shader from a `vec3` to a `vec4` as the spec states

> If the vertex shader has fewer components, the extra components are discarded.

![vertex_input_data_processing_example_b_format](../images/vertex_input_data_processing_example_b_format.png)

### Example C - non-interleaved

> Sometimes data is not interleaved, in this case, there might have the following

Reviewer: ```suggestion
Sometimes data is not interleaved, in this case, you might have the following
```
